doc/md/writing-motoko/incomplete-code.md

```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 27
+---
+# Writing incomplete code
+
+## Overview
+
+In the midst of writing a program, you may want to run an incomplete version or a version where one or more execution paths are either missing or simply invalid.
+
+To accommodate these situations, Motoko provides the `xxx`, `nyi` and `unreachable` functions from the base `Prelude` library, explained below. Each is a simple wrapper around a more [general trap mechanism](traps.md).
+
+## Short-term holes
+
+
+Assuming that one has imported the prelude as follows:
+
+``` motoko name=prelude
+import P "mo:base/Prelude";
+```
+
+The developer can fill any missing expression with the following:
+
+``` motoko include=prelude
+P.xxx()
+```
+
+The expression `xxx()` has type `None`, which is a subtype of every other type. This means the expression can plug any short-term gap in your code. 
+`xxx()` will trap when executed.
+
+## Long-term holes
+
+By convention, longer-term holes can be considered "not yet implemented" (`nyi`) features, and marked as such with a similar function from the Prelude module:
+
+``` motoko include=prelude
+P.nyi()
+```
+Like `xxx()`, `nyi()` has type `None`, thus any other type, and will trap on execution.
+## `unreachable` code paths
```

Comment:
```suggestion
## Unreachable code

```

Q: the page starts with `you` but then talks about the developer - should it stick with `you` or is this switching ok? Up to you.

Reply:
Good catch, it should all be `you`. Updated to fix.